### PR TITLE
Try to avoid fetch errors in spack CI

### PR DESCRIPTION
**Pull request by aurianer**

### CODE REVIEW

File: ci/docker/build.Dockerfile
Comment (albestro) [inline ci/docker/build.Dockerfile:81]: CI failed because `spack:` section is the root for environment configuration, but this is going to change `config.yaml` in the user scope, which is not part of the environment.

I think you can try something like

```
spack config --scope user add 'config:url_fetch_method:curl'
```

**Comment by albestro** [comment]
AFAICT, in the CI we are experiencing different problems:
1. Error: FetchError: All fetchers failed
2. Error: TimeoutError: The read operation timed out
3. Error: ChecksumError: sha256 checksum failed

About 1, which is the one that this PR is targeting, it happens very often (non complete list)
- https://gitlab.com/cscs-ci/eth-cscs/DLA-Future/-/jobs/3123498727
- https://gitlab.com/cscs-ci/eth-cscs/DLA-Future/-/jobs/3131073830
- https://gitlab.com/cscs-ci/eth-cscs/DLA-Future/-/jobs/3135172798
- https://gitlab.com/cscs-ci/eth-cscs/DLA-Future/-/jobs/3118583838
- https://gitlab.com/cscs-ci/eth-cscs/DLA-Future/-/jobs/3118583840
- https://gitlab.com/cscs-ci/eth-cscs/DLA-Future/-/jobs/3111758795

and it seems to always happen with `ncurses`. As a spack core maintainer was suggesting on the spack slack channel, it can also be a problem with the url origin. One solution can be to use `ncurses` as external package?

About the other two, I've seen them just once ([2](https://gitlab.com/cscs-ci/eth-cscs/DLA-Future/-/jobs/3125181116) and [3](https://gitlab.com/cscs-ci/eth-cscs/DLA-Future/-/jobs/3130491797)), and I speculate it can be related to some problem with the filesystem in the CI.

**Comment by albestro** [comment]
Apparently it happened again, still case 1 (ncurses)

https://gitlab.com/cscs-ci/eth-cscs/DLA-Future/-/jobs/3149394655

**Review by msimberg — APPROVED** [review]
Looks good to me. Have you checked that ncurses is really found and added to `packages.yaml` locally? Alternatively we can do one round of CI with a `RUN cat ~.spack/packages.yaml` step.

**Comment by albestro** [comment]
> Looks good to me. Have you checked that ncurses is really found and added to `packages.yaml` locally? Alternatively we can do one round of CI with a `RUN cat ~.spack/packages.yaml` step.

Not sure in which CI image I looked into, but by checking in `release-cpu-clang13-20/build:latest` I got

```yaml
root@845d113f65a4:/# spack config get packages | grep ncurses -A 2
  ncurses:
    externals:
    - spec: ncurses@6.3.20211021+symlinks+termlib abi=6
      prefix: /usr
```

**Comment by msimberg** [comment]
> > Looks good to me. Have you checked that ncurses is really found and added to `packages.yaml` locally? Alternatively we can do one round of CI with a `RUN cat ~.spack/packages.yaml` step.
> 
> Not sure in which CI image I looked into, but by checking in `release-cpu-clang13-20/build:latest` I got
> 
> ```yaml
> root@845d113f65a4:/# spack config get packages | grep ncurses -A 2
>   ncurses:
>     externals:
>     - spec: ncurses@6.3.20211021+symlinks+termlib abi=6
>       prefix: /usr
> ```

That looks like it should be enough, thanks for checking!

**Review by msimberg — CHANGES_REQUESTED** [review]
Let's minimize this to only having `ncurses` as an external and merge this after that. Thanks!

**Review by albestro — APPROVED** [review]

**Review by msimberg — APPROVED** [review]

**Comment by msimberg** [comment]
I'm waiting to see if CI runs normally in this job: https://gitlab.com/cscs-ci/eth-cscs/DLA-Future/-/pipelines/671873439 (different PR). If yes I'll merge this.

**Pull request merged**
